fasttraveler/fasttraveler.lua

```diff
@@ -0,0 +1,129 @@
+--[[
+* rMod - Copyright (c) 2018 atom0s [atom0s@live.com]
+*
+* This work is licensed under the Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International License.
+* To view a copy of this license, visit http://creativecommons.org/licenses/by-nc-nd/4.0/ or send a letter to
+* Creative Commons, PO Box 1866, Mountain View, CA 94042, USA.
+*
+* By using rMod, you agree to the above license and its terms.
+*
+*      Attribution - You must give appropriate credit, provide a link to the license and indicate if changes were
+*                    made. You must do so in any reasonable manner, but not in any way that suggests the licensor
+*                    endorses you or your use.
+*
+*   Non-Commercial - You may not use the material (rMod) for commercial purposes.
+*
+*   No-Derivatives - If you remix, transform, or build upon the material (rMod), you may not distribute the
+*                    modified material. You are, however, allowed to submit the modified works back to the original
+*                    rMod project in attempt to have it added to the original project.
+*
+* You may not apply legal terms or technological measures that legally restrict others
+* from doing anything the license permits.
+*
+* No warranties are given.
+]]--
+
+_addon.author   = 'Kender';
+_addon.name     = 'fasttraveler';
+_addon.version  = '1.0';
+_addon.description = 'Allows fast travel from inside dungeons, caves, buildings, etc.'
+
+require 'common';
+
+----------------------------------------------------------------------------------------------------
+-- Fast traveler Variables
+----------------------------------------------------------------------------------------------------
+local fasttraveler = { };
+fasttraveler.address = 0;
+
+----------------------------------------------------------------------------------------------------
+-- func: set_fast_traveler
+-- desc: Enables or disables Fast traveler.
+----------------------------------------------------------------------------------------------------
+local function set_fast_traveler(enabled)
+    -- Ensure the required address was found..
+    if (fasttraveler.address == 0) then
+        return;
+    end
+
+    -- Prepare the write value..
+    local v = string.char(0xE8, 0xD6, 0x5B, 0xD4, 0xFF);
```

Comment:
For these, you don't need to convert between char arrays like this. Instead you can write the full list of bytes as a table if the write_array function. Which can be used like:

```lua
hook.memory.write_array(fasttraveler.address, { 0xE8, 0xD6, 0x5B, 0xD4, 0xFF });
```